web-responsive/components/navbar.tsx

```diff
@@ -1,136 +1,248 @@
+            <Typography
+              sx={{
+                color: "#0E204E",
+                fontFamily: "Playfair Display",
+                fontSize: 20,
+                fontStyle: "normal",
+                fontWeight: 700,
+                lineHeight: "normal",
+                width: "92px",
+              }}
+            >
+              Pineapple
+              Island
+            </Typography>
+          </Box>
+          <Box>
+            <Hamburger
+              toggled={isDrawerOpen}
+              toggle={setIsDrawerOpen}
+              size={20}
+            />
 
-            <MenuItem onClick={handleClose}>
-              <Link href="#whatWeDoPage" underline="none" className={styles.content}>
-                {'What we do'}
-              </Link>
-            </MenuItem>
+            <Drawer
+              anchor="right"
+              open={isDrawerOpen}
+              onClose={() => setIsDrawerOpen(false)}
+            >
+              <Box
+                sx={{
+                  display: "flex",
+                  alignItems: "center",
+                  justifyContent: "flex-start",
+                  padding: 1,
+                }}
+              >
+                <IconButton onClick={handleClose}>
+                  <CloseIcon />
+                </IconButton>
+              </Box>
 
-            <MenuItem onClick={handleClose}>
-              <Link href="#projectPage" underline="none" className={styles.content}>
-                {'Project'} 
-              </Link>
-            </MenuItem>
-          </Menu>
-        </Grid>
-      </Grid>
-    </Grid>
+              <Box
+                sx={{
+                  paddingX: 1.25,
+                  display: "flex",
+                  flexDirection: "column",
+                  alignItems: "flex-end",
+                }}
+              >
+                <MenuItem onClick={handleClose}>
+                  <Link
+                    href="#aboutPage"
+                    underline="none"
+                    sx={{
+                      color: "#232323",
+                      fontFamily: "Manrope",
+                      fontSize: 20,
+                      fontStyle: "normal",
+                      fontWeight: 400,
+                      lineHeight: "normal",
+                    }}
+                  >
+                    About Us
+                  </Link>
+                </MenuItem>
+
+                <MenuItem onClick={handleClose}>
+                  <Link
+                    href="#whatWeDoPage"
+                    underline="none"
+                    sx={{
+                      color: "#232323",
+                      fontFamily: "Manrope",
+                      fontSize: 20,
+                      fontStyle: "normal",
+                      fontWeight: 400,
+                      lineHeight: "normal",
+                    }}
+                  >
+                    What we do
+                  </Link>
+                </MenuItem>
```

Comment:
ตรงนี้เข้าใจว่าน่าจะเป็นการสั่งให้ตัว Drawer ปิดหลังจากที่กดปุ่มแล้วใช่ไหม ถ้าใช่พี่ว่าตรงนี้น่าจะรวมทำเป็น function เดียวได้ อาจจะให้มันกดแล้วไปที่ section นั้นๆพร้อมกับปิดตัว Drawer ไปด้วยเลย ลองทำดูๆ

Reply:
หมายถึงว่า handleClose ไม่จำเป็นจต้องใช้ก็ได้ใช่มั้ยครับ

Reply:
หมายถึงว่าเราอาจจะรวมการ scroll ไปยัง section ต่างๆกับการปิดตัว Drawer ไว้ที่ function เดียวกันได้ครับ
not sure if this work:
`const scrollToSection  = (section: string) => { router.push('#${section}'); setIsOpen(false) }`